src/cascadia/TerminalApp/TerminalTab.h

```diff
@@ -93,7 +94,7 @@ namespace winrt::TerminalApp::implementation
         std::optional<winrt::Windows::UI::Color> _themeTabColor{};
         std::optional<winrt::Windows::UI::Color> _runtimeTabColor{};
 
-        bool _focused{ false };
+        winrt::Windows::UI::Xaml::FocusState _focusState{ winrt::Windows::UI::Xaml::FocusState::Unfocused };

```

Comment:
I think I like this change. But why did we have to do it?

Reply:
Instead of keeping the `Focus` as a boolean, I chose to copy the way XAML controls handles focus with `FocusStates` and bring it to `ITab` and all its implementers because I felt it would be good in general to know how the tab content is receiving focus. 

For example, having the ability to tell the Settings UI tab how it was focused (through a pointer interaction vs keyboard) could let us determine if we want to show a focus indicator outline and other a11y functionality as focus is gained. 

Unfortunately, we currently don't really plumb through our code the FocusState that started an action. That's why I've left all the focus calls as `Focus(FocusState::Programmatic)`.